zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AsterixQueryGenerator.scala

```diff
@@ -0,0 +1,270 @@
+package edu.uci.ics.cloudberry.zion.model.impl
+
+import edu.uci.ics.cloudberry.zion.model.datastore.{IQLGenerator, QueryParsingException}
+import edu.uci.ics.cloudberry.zion.model.schema._
+
+
+/**
+  * Defines constant strings for query languages supported by AsterixDB
+  */
+trait TypeImpl {
+
+  val aggregateFuncMap: Map[AggregateFunc, String]
+
+  def getAggregateStr(aggregate: AggregateFunc): String = {
+    aggregateFuncMap.get(aggregate) match {
+      case Some(impl) => impl
+      case None => throw new QueryParsingException(s"No implementation is provided for aggregate function ${aggregate.name}")
+    }
+  }
+
+  val datetime: String = "datetime"
+  val round: String = "round"
```

Comment:
Then I will suggest to remove the implementation and implement the `val` separately in both classes. Though the implementation is exactly the same, it make the `trait` more safe.